cmd/tusd/cli/hooks.go

```diff
@@ -67,14 +70,67 @@ func invokeHookSync(typ HookType, info tusd.FileInfo, captureOutput bool) ([]byt
 		logEv("UploadTerminated", "id", info.ID)
 	}
 
-	if !Flags.HooksInstalled {
+	if !Flags.FileHooksInstalled && !Flags.HttpHooksInstalled {
 		return nil, nil
 	}
-
 	name := string(typ)
 	logEv("HookInvocationStart", "type", name, "id", info.ID)
 
-	cmd := exec.Command(Flags.HooksDir + "/" + name)
+	output := []byte{}
+	err := error(nil)
+
+	if Flags.FileHooksInstalled {
+		output, err = invokeFileHook(name, typ, info, captureOutput)
+	}
+
+	if Flags.HttpHooksInstalled {
+		output, err = invokeHttpHook(name, typ, info, captureOutput)
+	}
+
+	return output, err
+}
+
+func invokeHttpHook(name string, typ HookType, info tusd.FileInfo, captureOutput bool) ([]byte, error) {
+	url := Flags.HttpHooksEndpoint
+	jsonInfo, err := json.Marshal(info)
+
+	req, err := http.NewRequest("POST", url, bytes.NewReader(jsonInfo))
+	req.Header.Set("Upload-State", name)
+	req.Header.Set("Content-Type", "application/json")
+	if err != nil {
+		return nil, err
+	}
+
+	//retry 3 times at most.
+	client := pester.New()
+	client.KeepLog = true
+	client.Backoff = func(_ int) time.Duration {
+		return time.Duration(Flags.HttpHooksBackoff) * time.Second
+	}
+	client.MaxRetries = Flags.HttpHooksRetry
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return nil, err
+	}
+	defer resp.Body.Close()
+
+	response := []byte(resp.Status)
+	if resp.StatusCode >= 400 {
+		err := errors.New("Invalid Response Code")
+		return response, err
+	}
+
+	//if capture output is true, this function will return both the response and rthe err. Else it will return only the nil and err
+	if !captureOutput {
```

Comment:
Wait I think this is wrong. It should just be `if captureOutput` not `if !captureOutput` right?